File: FE/src/constants/password-regex.ts
import { config } from "@/config/config";

// Password must be 8-200 characters, with at least one lowercase, uppercase, digit, and special character.
export const PASSWORD_FULL_REGEX = new RegExp(
  `^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^A-Za-z\\d])[A-Za-z\\d\\S]{${config.MIN_PASSWORD_LENGTH},${config.MAX_PASSWORD_LENGTH}}$`
);

// Only check for at least one lowercase, uppercase, digit, and special character.
// For use cases such as form validation, where the validator have different messages based on the validation error, use this instead of the full one.
export const PASSWORD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^A-Za-z\d])[A-Za-z\d\S]+$/;

File: BE/src/main/java/com/example/demo_springboot_api/common/constants/PasswordRegex.java
package com.example.demo_springboot_api.common.constants;

import com.example.demo_springboot_api.config.constants.UserInfoConfig;

public class PasswordRegex {
  // Password must be 8-200 characters, with at least one lowercase, uppercase, digit, and special
  // character.
  public static final String PASSWORD_FULL_REGEX =
      "^(?=.{%d,%d}$)[a-z0-9]+(?:\\.[a-z0-9]+)*$"
          .formatted(UserInfoConfig.MIN_USER_CODE_LENGTH, UserInfoConfig.MAX_USER_CODE_LENGTH);

  // Only check for at least one lowercase, uppercase, digit, and special character.
  // For use cases such as form validation, where the validator have different messages based on the
  // validation error, use this instead of the full one.
  public static final String PASSWORD_REGEX =
      "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^A-Za-z\\d])[A-Za-z\\d\\S]+$";
}
